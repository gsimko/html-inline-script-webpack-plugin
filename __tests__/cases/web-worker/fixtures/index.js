const worker = new Worker(new URL(/* webpackChunkName: 'test.worker' */'./worker.js', import.meta.url));

let result;

worker.onmessage = function (event) {
  if (!result) {
    result = document.createElement('div');
    result.setAttribute('id', 'result');

    document.body.append(result);
  }

  const record = document.createElement('pre');
  record.innerHTML = JSON.stringify(event.data);

  result.append(record);
};

window.addEventListener('load', () => {
  const button = document.getElementById('button');

  button.addEventListener('click', () => {
    worker.postMessage({ postMessage: true });
  });
});
