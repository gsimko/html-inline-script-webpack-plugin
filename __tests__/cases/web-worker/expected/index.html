<!doctype html><html lang="en"><head><meta charset="utf-8"/><meta http-equiv="Content-Type" content="text/html; charset=utf-8"/><meta name="language" content="English"/><meta http-equiv="X-UA-Compatible" content="IE=edge"/><meta name="viewport" content="minimum-scale=1,initial-scale=1,width=device-width,shrink-to-fit=no"/><title>webpack test</title><script defer="defer">(()=>{var e={m:{},u:e=>"test.worker.js",o:(e,t)=>Object.prototype.hasOwnProperty.call(e,t),p:"./"};e.b=document.baseURI||self.location.href;const t=new Worker(new URL(e.p+e.u(912),e.b));let n;t.onmessage=function(e){n||(n=document.createElement("div"),n.setAttribute("id","result"),document.body.append(n));const t=document.createElement("pre");t.innerHTML=JSON.stringify(e.data),n.append(t)},window.addEventListener("load",(()=>{document.getElementById("button").addEventListener("click",(()=>{t.postMessage({postMessage:!0})}))}))})();</script></head><body><p>This is minimal code to demonstrate webpack usage</p><button id="button">Run Action</button></body></html>